pkg/controller/minio.go

```diff
@@ -188,6 +188,8 @@ func (c *Controller) checkOperatorCaForTenant(ctx context.Context, tenant *minio
 		}
 		return false, fmt.Errorf("'public.crt' in '%s/%s' secret changed, updating '%s/%s' secret", miniov2.GetNSFromFile(), OperatorCATLSSecretName, tenant.Namespace, OperatorCATLSSecretName)
 	}
+	// Reload CA TLS certificates
```

Comment:
I think we can save `OperatorCATLSSecretName`'s `public.crt`. Check if before and after. If it's the same, we don't need reload. And how about we set the args as `reload ...bool`.

Reply:
And we must can't to update the secret if `tenant`'s namespace are equal with minio-operator's namespace. If we do that, the data will be lose!

Reply:
And we must can't update the `OperatorCATLSSecretName` if tenant's namespace are equal with minio-operator's namespace. The data will be lose. `"tls.crt","ca.crt"`

Reply:
> I think we can save `OperatorCATLSSecretName`'s `public.crt`. Check if before and after. If it's the same, we don't need reload. And how about we set the args as `reload ...bool`.

I am not sure on this one, there is multiple calls to the getTransport method per synchandler iteration, returning a cached transport reduces the risk of k8s API throttling and simply there is no need to read the secret unless we know it was modified for sure.

Reply:
> And we must can't update the `OperatorCATLSSecretName` if tenant's namespace are equal with minio-operator's namespace. The data will be lose. `"tls.crt","ca.crt"`

This is fine, `operator-ca-tls` secret is only modified manually specifically by the administrator, whatever is on this secret overwrites whatever was before because is an explicit action.

Reply:
> > I think we can save `OperatorCATLSSecretName`'s `public.crt`. Check if before and after. If it's the same, we don't need reload. And how about we set the args as `reload ...bool`.
> 
> I am not sure on this one, there is multiple calls to the getTransport method per synchandler iteration, returning a cached transport reduces the risk of k8s API throttling and simply there is no need to read the secret unless we know it was modified for sure.

What I mean is that in this function processing

Reply:
> > And we must can't update the `OperatorCATLSSecretName` if tenant's namespace are equal with minio-operator's namespace. The data will be lose. `"tls.crt","ca.crt"`
> 
> This is fine, `operator-ca-tls` secret is only modified manually specifically by the administrator, whatever is on this secret overwrites whatever was before because is an explicit action.

I know it, I mean, if minio-operator's namespace are equal with tenant's namespace. We will update the secret. But we only update with `public.crt`.

Reply:
> 



> > > And we must can't update the `OperatorCATLSSecretName` if tenant's namespace are equal with minio-operator's namespace. The data will be lose. `"tls.crt","ca.crt"`
> > 
> > 
> > This is fine, `operator-ca-tls` secret is only modified manually specifically by the administrator, whatever is on this secret overwrites whatever was before because is an explicit action.
> 
> I know it, I mean, if minio-operator's namespace are equal with tenant's namespace. We will update the secret. But the operator only update the secret with `public.crt` for now.

Well, is not a good practice to mixup Operator and tenant in the same namespace, they should be on different namespaces.

On the other one, you are right, only the `public.crt` is being updated to tenants, fixing that on a following commit